### PR TITLE
Workaround for wrangler

### PR DESCRIPTION
**Pull request by yaojiefeng**
This is a workaround for wrangler FLL. Currently applied to the following directives: Delete a Column, Set a Column, Send to Error, Mask, Filter, Fill-Null-Or-Empty
Change Data Type

### CODE REVIEW

File: wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java [hunk 1]
```diff
@@ -511,14 +471,89 @@ private Object getValue(StructuredRecord input, String fieldName) {
...
+    return true;
+  }
+
+  /**
+   * Validates input schema.
+   *
+   * @param inputSchema configured for the plugin
+   * @param collector failure collector
+   */
+  private void validateInputSchema(Schema inputSchema, FailureCollector collector) {
```
Comment (chtyim) [inline wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java]: `InputSchema` should be `@Nullable`

**Review by chtyim — APPROVED** [review]
Just one minor comment

File: wrangler-api/src/main/java/io/cdap/wrangler/api/Directive.java [hunk 2]
```diff
@@ -115,4 +116,13 @@
    * @see io.cdap.wrangler.api.parser.TokenType
    */
   UsageDefinition define();
+
```
Comment (nitinmotgi) [inline wrangler-api/src/main/java/io/cdap/wrangler/api/Directive.java:119]: Should not be added to this interface. Could we add another interface named Mutation that has this method. Only directives that implement that interface will that. This is very clunky.

File: wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java [hunk 3]
```diff
@@ -511,14 +471,89 @@ private Object getValue(StructuredRecord input, String fieldName) {
...
    * Retrieves the base url from the context and appends method to value to the final url.
    *
    * @param method to be invoked.
    * @return fully formed url to the method.
    */
-  private URL getDPServiceURL(String method) throws URISyntaxException, MalformedURLException {
-    URL url = getContext().getServiceURL(Contexts.SYSTEM, APPLICATION_NAME, SERVICE_NAME);
+  private URL getDPServiceURL(ServiceDiscoverer discoverer,
+                              String method) throws URISyntaxException, MalformedURLException {
+    URL url = discoverer.getServiceURL(Contexts.SYSTEM, APPLICATION_NAME, SERVICE_NAME);
```
Comment (nitinmotgi) [inline wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java:556]: what if discoverer is null ?

Comment (yaojiefeng) [reply wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java:556]: The discoverer here is the StageContext, it cannot be null

File: wrangler-api/src/main/java/io/cdap/wrangler/api/Directive.java (hunk 2 above)
Comment (yaojiefeng) [reply wrangler-api/src/main/java/io/cdap/wrangler/api/Directive.java:119]: Addressed in https://github.com/data-integrations/wrangler/pull/365

**Pull request merged**